plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt

```diff
@@ -308,6 +345,71 @@ class SentryPlugin : Plugin<Project> {
 
             project.installDependencies(extension)
         }
+
+        project.plugins.withType(JavaBasePlugin::class.java) {
+            if (project.pluginManager.hasPlugin("com.android.application")) {
+                // AGP also applies JavaBasePlugin, but since we have a separate setup for it,
+                // we just bail here
+                return@withType
+            }
+            if (configuredForJavaProject.getAndSet(true)) {
+                logger.info { "The Sentry Gradle plugin was already configured" }
+                return@withType
+            }
+
+            val javaExtension = project.extensions.getByType(JavaPluginExtension::class.java)
+
+            val sentryResDir = project.layout.buildDirectory.dir("generated${sep}sentry")
+            javaExtension.sourceSets.getByName("main").resources { sourceSet ->
```

Comment:
`m` Should we also allow for devs to configure the source set they want?
Maybe even support multiple and have `listOf("main")` as default?
This could be a separate issue as well.

Reply:
hmm, but what would be the usecase? All the sourcesets get packaged into the final jar anyway, and that's what we're interested in

Reply:
I was thinking about people renaming the source sets because they're following some conventions or have to support some legacy code base where names are different.